taskotron_python_versions/unversioned_shebangs.py

```diff
@@ -0,0 +1,101 @@
+import sys
+import libarchive
+
+from .common import log, write_to_artifact
+
+MESSAGE = """These RPMs contains problematic shebang in some of the scripts:
+{}
+This is discouraged and should be avoided. Please check the shebangs
+and use either `#!/usr/bin/python2` or `#!/usr/bin/python3`.
+"""
+
+# TODO: update to real doc relevant to shebangs
+INFO_URL = ' https://pagure.io/packaging-committee/issue/698'
+
+FORBIDDEN_SHEBANGS = ['#!/usr/bin/python', '#!/usr/bin/env python']
+
+if sys.version > '3':
+    def to_bytes(s):
+        return str.encode(s)
+else:
+    def to_bytes(s):
+        return s
+
+
+def matches(line, query):
+    return line == query or line.startswith(query + b' ')
+
+
+def get_problematic_files(archive, query):
+    """Search for the files inside archive with the first line
+    matching given query.
+    """
+
+    problematic = set()
+    with libarchive.file_reader(archive) as a:
+        for entry in a:
+            try:
+                first_line = next(entry.get_blocks(), '').splitlines()[0]
+            except IndexError:
+                continue
+            if matches(first_line, to_bytes(query)):
+                problematic.add(entry.pathname.lstrip('.'))
+
+    return problematic
+
+
+def get_scripts_summary(package):
+    """Collect problematic scripts data for given RPM package.
+    Content of archive is processed only package require
+    unversioned python binary.
+    """
+    scripts_summary = {key: set() for key in FORBIDDEN_SHEBANGS}
+
+    for key in scripts_summary.keys():
```

Comment:
what if you do:

```python
for shebang in FORBIDDEN_SHEBANGS:
   ...
```

And use a `defaultdict(set)` for `scripts_summary`? It would be more consistent with other tasks made by @irushchyshyn. And IMHO more readable as well.